bokehjs/src/lib/models/graphs/graph_hit_test_policy.ts

```diff
@@ -283,3 +283,85 @@ export class EdgesAndLinkedNodes extends GraphHitTestPolicy {
     return !edge_inspection.is_empty()
   }
 }
+
+export namespace NodesAndAdjacentNodes {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = GraphHitTestPolicy.Props
+}
+
+export interface NodesAndAdjacentNodes extends NodesAndAdjacentNodes.Attrs {}
+
+export class NodesAndAdjacentNodes extends GraphHitTestPolicy {
+  override properties: NodesAndAdjacentNodes.Props
+
+  constructor(attrs?: Partial<NodesAndAdjacentNodes.Attrs>) {
+    super(attrs)
+  }
+
+  hit_test(geometry: Geometry, graph_view: GraphRendererView): HitTestResult {
+    return this._hit_test(geometry, graph_view, graph_view.node_view)
+  }
+
+  get_adjacent_nodes(node_source: ColumnarDataSource, edge_source: ColumnarDataSource, mode: string): Selection {
+    let selected_node_indices = []
+    if (mode == "selection") {
+      selected_node_indices = node_source.selected.indices.map((i) => node_source.data.index[i])
+    } else if (mode == "inspection") {
+      selected_node_indices = node_source.inspected.indices.map((i) => node_source.data.index[i])
+    }
+    const adjacent_nodes = []
+    const selected_nodes = []
+    for (let i = 0; i < edge_source.data.start.length; i++) {
+      if (contains(selected_node_indices, edge_source.data.start[i])) {
+        adjacent_nodes.push(edge_source.data.end[i])
+        selected_nodes.push(edge_source.data.start[i])
+      }
+      if (contains(selected_node_indices, edge_source.data.end[i])) {
+        adjacent_nodes.push(edge_source.data.start[i])
+        selected_nodes.push(edge_source.data.end[i])
+      }
+    }
+    for (let i = 0; i < selected_nodes.length; i++)
+      adjacent_nodes.push(selected_nodes[i])
+
+    const adjacent_node_indices = uniq(adjacent_nodes).map((i) => indexOf(node_source.data.index, i))
+    return new Selection({indices: adjacent_node_indices})
+  }
+
+  do_selection(hit_test_result: HitTestResult, graph: GraphRenderer, final: boolean, mode: SelectionMode): boolean {
+    if (hit_test_result == null)
+      return false
+
+    const initial_node_selection = graph.node_renderer.data_source.selected
+    initial_node_selection.update(hit_test_result, final, mode)
+
+    const final_node_selection = graph.node_renderer.data_source.selected
```

Comment:
I had to squint at this for a second. I think assigning `graph.node_renderer.data_source.selected` to two different names, which are then used to mutate the same value, is confusing. I'd prefer to re-work this with a single
```
const node_selection = graph.node_renderer.data_source.selected
``` 
at the top that is used throughout

Reply:
Or if you want to be more compact
```
const {selected} = graph.node_renderer.data_source
```
that is used throughout for updates

Reply:
Sure, I can change this to be more readable!